typify-impl/src/lib.rs

```diff
@@ -1098,6 +1262,496 @@ impl<'a> TypeNewtype<'a> {
+                    base_schema,
+                    &definition_schema,
+                    base_path,
+                    base_id,
+                    external_references,
+                    false,
+                );
+            }
+        } else {
+            // Handle external references
+            let base_id = base_id
+                .as_ref()
+                .expect("missing 'id' attribute in schema definition"); // Ensure base_id is present
+            let id = Iri::new(base_id).unwrap(); // Create an IRI from the base ID
+            let reff = Iri::new(&reference).unwrap(); // Create an IRI from the reference
+            let fragment = reff
+                .fragment()
+                .as_ref()
+                .unwrap_or(&FragmentBuf::new("".to_string()).unwrap().as_fragment())
+                .to_string()
+                .split("/")
+                .filter_map(|s| (!s.is_empty()).then_some(s.to_string()))
+                .collect::<Vec<_>>(); // Process the fragment part of the reference
+            let relpath =
+                diff_paths(reff.path().as_str(), id.path().parent_or_empty().as_str()).unwrap(); // Determine the relative path
+            let file_path = base_path.parent().unwrap().join(&relpath); // Construct the file path
+            let content = std::fs::read_to_string(&file_path).expect(&format!(
+                "Failed to open input file: {}",
+                &file_path.display()
+            )); // Read the file content
+
+            let root_schema = serde_json::from_str::<RootSchema>(&content)
+                .expect("Failed to parse input file as JSON Schema"); // Parse the file content as JSON Schema
+            let definition_schema = fetch_defenition(&root_schema, &reference, &fragment); // Fetch the external schema definition
+            let key = RefKey::Def(reference.clone());
+            if external_references.contains_key(&key) {
+                continue; // Skip if the reference already exists in the map
+            } else {
+                let s_id = get_schema_id(&root_schema.schema); // Get the schema ID
+
+                // Insert the reference into the map and recursively fetch external definitions
+                external_references.insert(
+                    key,
+                    (definition_schema.clone(), file_path.clone(), s_id.clone()),
+                );
+                fetch_external_definitions(
+                    &root_schema,
+                    &definition_schema,
+                    &file_path,
+                    &s_id,
+                    external_references,
+                    false,
+                )
+            }
+        }
+    }
+}
+
+fn fetch_defenition(
+    base_schema: &RootSchema,
+    reference: &String,
+    fragment: &Vec<String>,
+) -> Schema {
+    if fragment.is_empty() {
+        return Schema::Object(base_schema.schema.clone());
+    }
+    let definition_schema = if fragment[0] == "definitions" {
```

Comment:
```suggestion
    let definition_schema = if ["definitions", "$defs"].contains(&fragment[0].as_str()) {
```
This should match both "definitions" and "$defs" as "https://json-schema.org/draft/2020-12/schema" [replaced "definitions" with "$defs"](https://json-schema.org/draft/2020-12/json-schema-core#appendix-G-2.6.1.2)